doc/Language/variables.pod6

```diff
@@ -498,6 +399,212 @@ augment slang Regex {  # derive from $~Regex and then modify $~Regex
 }
 =end code
 
+
+=head1 Item assignment, list assignment, and destructuring assignment
+
+There are three types of variable assignment, I<item assignment>, I<list
+assignment>, and I<destructuring assignment>.
+
+An item assignment copies a single value from the right-hand side into a Scalar
+variable on the left.  For example:
+
+    my $a;
+    $a = 1; # Item assignment of an Int to a Scalar in $a;
+    say $a; say $a.VAR.^name; # OUTPUT: «1␤Scalar␤»
+
+    my $b = (1, 2, 3); # Item assignment of a List to a Scalar in $a;
+    say $b; say $b.VAR.^name; # OUTPUT: «(1 2 3)␤Scalar␤»
+
+Any assignment with anything other than a simple Scalar variable to the left
+of the C<=> operator is a list assignment. This includes a Scalar variable wrapped
+in parentheses, such as C<my ($a)>
+
+The effect of a list assignment is determined by the variable(s) and sigil(s) to
+the left of the C<=> (the "left hand side").  Specifically, a variable with a
+C<$> sigil or without a sigil takes one value from the right hand side and
+stores it in a Scalar; a C<@>-sigiled variable takes all remaining variables
+from the right hand side and copies them into a L<Positional|/type/Positional>
+(usually an C<Array>); and a C<%>-sigiled variable takes all remaining values on
+the right hand side and copies them into an L<Associative|/type/Associative>
+(usually a C<Hash>).  If those values are C<Pairs>, then the C<Pair>'s key and
+value will become the C<Hash>'s key and value. If the right-hand-side values are
+anything other than C<Pair>s, then the C<%>-sigiled variable will take
+alternating values as keys and values.  Here are some examples:
+
+
+    my ($a) = (1, 2, 3);      # list assignment to a Scalar;
+    say $a; say $a.VAR.^name; # OUTPUT:  «1␤Scalar␤»
+
+    my @b = (1, 2, 3);        # list assignment to a Positional
+    say @b;                   # OUTPUT:  «[1 2 3]␤»
+
+    my ($c, $d, %e) = (1, 2, :k(3)); # list assignment to two Scalars and a Positional
+    say $c; say $d; say %e;          # OUTPUT:  «1␤2␤{k => 3}]␤»
+
+    # If there are fewer values than LHS variables, some will get Any
+    my ($f, $g, $h, $i) = (1, 2, 3); # list assignment to four Scalars
+    say $f; say $g; say $h; say $i;  # OUTPUT:  «1␤2␤3␤(Any)␤»
+
+    # @-sigiled variables take all remaining values
+    my ($j, @k, $l) = (1, 2, 3);
+    say $j; say @k; say $l;      # OUTPUT:  «1␤[2 3]␤(Any)␤»
+
+    # so do %-sigiled variables
+    my ($m, %n, $o) = (1, 2, 3);
+    say $m; say %n; say $o;      # OUTPUT:  «1␤{2 => 3}␤(Any)␤»
+
+    # if a %-sigiled variable can't construct a Hash, you get an error
+    try my ($q, %r) = (1, 2);
+    say $!;
+    # OUTPUT: «Odd number of elements found where hash initializer expected:␤
+    #          Only saw: 2»
+
+
+Both item assignment and list assignment use the equal sign (C<=>) as their
```

Comment:
Nit: repetition of both (actually carried over from original text, but as long as you're changing stuff...)